apps/state/lib/state/stop.ex

```diff
@@ -215,7 +230,13 @@ defmodule State.Stop do
         end
       end)
 
-    Enum.uniq_by(stops, & &1.id)
+    search_results
+    |> Enum.to_list()
+    |> Enum.reduce(all(), fn results, acc ->
```

Comment:
1. You don't need `Enum.to_list`, as `Enum.reduce` can work against anything enumerable (including streams).
1. This will always ask for all the stops, even when there are filters.